keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ExportService.kt

```diff
@@ -0,0 +1,324 @@
+package com.netflix.spinnaker.keel.services
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.netflix.spinnaker.keel.api.Constraint
+import com.netflix.spinnaker.keel.api.ResourceKind
+import com.netflix.spinnaker.keel.api.artifacts.DebianArtifact
+import com.netflix.spinnaker.keel.api.plugins.ResourceHandler
+import com.netflix.spinnaker.keel.api.plugins.supporting
+import com.netflix.spinnaker.keel.core.api.DependsOnConstraint
+import com.netflix.spinnaker.keel.core.api.ManualJudgementConstraint
+import com.netflix.spinnaker.keel.core.api.SubmittedDeliveryConfig
+import com.netflix.spinnaker.keel.core.api.SubmittedEnvironment
+import com.netflix.spinnaker.keel.core.api.SubmittedResource
+import com.netflix.spinnaker.keel.front50.Front50Service
+import com.netflix.spinnaker.keel.front50.model.BakeStage
+import com.netflix.spinnaker.keel.front50.model.FindImageStage
+import com.netflix.spinnaker.keel.front50.model.Pipeline
+import com.netflix.spinnaker.keel.front50.model.Stage
+import com.netflix.spinnaker.keel.front50.model.findPipelineWithDeployForCluster
+import kotlinx.coroutines.runBlocking
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class ExportService(
+  private val handlers: List<ResourceHandler<*, *>>,
+  private val front50Service: Front50Service
+) {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  companion object {
+    val EXPORTABLE_PIPELINE_PATTERNS = listOf(
```

Comment:
Thank you @michaelgalloway for your research and coming up with this list of patterns! 🙏